src/components/common/ShareCollectionModal/ShareCollectionModal.tsx

```diff
@@ -126,32 +132,20 @@ class ShareCollectionModal extends React.Component<Props, State> {
   };
 
   submit = async () => {
+    let collectionId = this.getCollectionId();
     // If the collection became private, only call the collectionSetPrivate mutation
     if (
       this.getCollection().isPublic &&
       this.state.isPublicOverride === false
     ) {
-      const Mutation = `
-        mutation setPrivate($collectionId: Int!) {
-          collectionSetPrivate(id: $collectionId)
-        }
-      `;
-      await collectionsGraphqlRequest(this.context, Mutation, {
-        collectionId: this.getCollectionId(),
-      });
+      await removeAllCollaborators(this.context, collectionId);
+      await setPrivate(this.context, collectionId);
```

Comment:
I think we should trigger a collection refetch right before closing the modal, as right now the collection is updated but the UI lags behind until a refresh.